crates/matrix-sdk-ui/src/timeline/builder.rs

```diff
@@ -239,13 +263,25 @@ impl TimelineBuilder {
                         RoomEventCacheUpdate::AddTimelineEvents { events, origin } => {
                             trace!("Received new timeline events.");
 
-                            inner.add_events_at(
-                                events,
-                                TimelineEnd::Back,
-                                match origin {
-                                    EventsOrigin::Sync => RemoteEventOrigin::Sync,
+                            // Special case for pinned events: when we receive new events what we'll do is
+                            // updated the cache for those events that are pinned and reload the
+                            // list.
+                            match &*focus.clone() {
+                                TimelineFocus::PinnedEvents { .. } => {
```

Comment:
So we ignore the events that appeared via `AddTimelineEvents` and just refresh the whole list? Might be good to make this clear in the comment as I didn't quite get it.

Reply:
Ok, I'll try to come up with a better comment to explain this.